app/code/core/Mage/Admin/Model/Observer.php

```diff
@@ -69,13 +73,14 @@ public function actionPreDispatchAdmin($observer)
                     $session->login($username, $password, $request);
                     $request->setPost('login', null);
                 }
-                if (!$request->getParam('forwarded')) {
```

Comment:
For the record I'd like to point out that adding this getInternallyForwarded/setInternallyForwarded is silly as the request object supports parameters that are set via code only. When you call setParam you are seting 'userland' parameters. If you want to get a parameter and be sure it was set via code and not the inbound requests you use getUserParam and NOT getParam like the original code had.